src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -78,10 +79,24 @@ public function handle()
      */
     protected function getFreshApplicationRoutes()
     {
-        return tap($this->getFreshApplication()['router']->getRoutes(), function ($routes) {
+        $application = $this->getFreshApplication();
+
+        $routes = tap($application['router']->getRoutes(), function ($routes) {
             $routes->refreshNameLookups();
             $routes->refreshActionLookups();
         });
+
+        if ($application->has('octane')) {
+            $warning = 'Running route:cache while running swoole decreases memory performance';
+            $this->components->warn($warning);
```

Comment:
You could put this all onto one line, I'm not sure there's a good reason to store the warning as a string first.